cmake/recipes/polysolve.cmake

```diff
@@ -0,0 +1,14 @@
+# PolySolve (https://github.com/polyfem/polysolve)
+# License: MIT
+
+if(TARGET polysolve)
+    return()
+endif()
+
+message(STATUS "Third-party: creating target 'polysolve'")
+
+# TODO: this requires a conflicting version of Eigen. Reenable when Eigen 3.4+ is available.
+set(POLYSOLVE_WITH_ACCELERATE OFF CACHE BOOL "Enable Apple Accelerate" FORCE)
```

Comment:
There is an issue with conflicting versions of Eigen when this option is enabled.

Reply:
Maybe @jdumas knows how?

Reply:
Why not let polysolve pull Eigen directly?

Reply:
I get this compilation error:
```
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.cpp:1:
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.hpp:3:
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/collisions/collision_constraints.hpp:3:
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/collision_mesh.hpp:5:
In file included from /Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/Core:329:
/Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/src/Core/IndexedView.h:148:14: error: data member instantiated with function type 'bool ()'
  ColIndices m_colIndices;
             ^
/Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.cpp:284:25: note: in instantiation of template class 'Eigen::IndexedView<const Eigen::Matrix<double, -1, -1, 0, 6, 12>, Eigen::ArithmeticSequence<long, long, long>, bool ()>' requested here
                        jac_Gamma_wrt_beta(
                        ^
/Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.cpp:285:63: error: no member named 'all' in namespace 'Eigen'; did you mean 'Eigen::indexing::all'?
                            Eigen::seqN(i, beta.size(), dim), Eigen::all)
                                                              ^~~~~~~~~~
                                                              Eigen::indexing::all
/Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/src/Core/ArithmeticSequence.h:250:30: note: 'Eigen::indexing::all' declared here
  using Eigen::placeholders::all;
                             ^
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.cpp:1:
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.hpp:3:
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/collisions/collision_constraints.hpp:3:
In file included from /Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/collision_mesh.hpp:5:
In file included from /Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/Core:290:
In file included from /Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/src/Core/DenseBase.h:662:
/Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/src/Core/util/../../plugins/IndexedViewMethods.h:63:73: error: no matching member function for call to 'ivcCol'
    return ConstReturnType(derived, derived.ivcRow(rowIndices), derived.ivcCol(colIndices));
                                                                ~~~~~~~~^~~~~~
/Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/src/Core/util/../../plugins/IndexedViewMethods.h:239:55: note: in instantiation of member function 'Eigen::DenseBase<Eigen::Matrix<double, -1, -1, 0, 6, 12>>::IndexedViewSelector<Eigen::ArithmeticSequence<long, long, long>, bool (), false, false, true>::run' requested here
  return IndexedViewSelector<RowIndices, ColIndices>::run(derived(), rowIndices, colIndices);
                                                      ^
/Users/zachary/.cache/CPM/ipc-toolkit/ff96ec46f901ecdd6203d75076174d996a5eae0d/src/ipc/friction/constraints/friction_constraint.cpp:284:25: note: in instantiation of function template specialization 'Eigen::DenseBase<Eigen::Matrix<double, -1, -1, 0, 6, 12>>::operator()<Eigen::ArithmeticSequence<long, long, long>, bool (), true>' requested here
                        jac_Gamma_wrt_beta(
                        ^
/Users/zachary/.cache/CPM/eigen/2e68522bbe453e136c9b281a06f112155cbeafd8/Eigen/src/Core/util/../../plugins/IndexedViewMethods.h:34:28: note: candidate template ignored: substitution failure [with Indices = bool ()]: function cannot return function type 'IvcColType<bool ()>' (aka 'bool ()')
inline IvcColType<Indices> ivcCol(const Indices& indices) const {
       ~~~~~~~~~~          ^
3 errors generated.
make[2]: *** [_deps/ipc-toolkit-build/CMakeFiles/ipc_toolkit.dir/src/ipc/friction/constraints/friction_constraint.cpp.o] Error 1
make[1]: *** [_deps/ipc-toolkit-build/CMakeFiles/ipc_toolkit.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Reply:
Created an Issue #240 and merging without the fix for now.